File: TypeScript/Map/remove-from-map.ts
function main(): void {
    // Create a new map.
    var randomMap: Map<string, string> = new Map([
        ["key1", "value1"],
        ["key2", "value2"]
    ])
    // Remove a key-value pair from the map.
    console.log(removeKeyValuePairFromMap(randomMap, "key1"))
}

main()

// Remove a key-value pair from the map and return the map.
function removeKeyValuePairFromMap(originalMap: Map<string, string>, key: string): Map<string, string> {
    originalMap.delete(key)
    return originalMap
}

File: Go/Map/remove-from-map.go
package main

import (
	"fmt"
)

func main() {
	// Create a simple map with a string as key and value as key.
	userGeneratedMap := map[string]string{"FirstName": "John", "LastName": "Doe"}
	// Remove a key-value pair from the map.
	fmt.Println(removeFromMap(userGeneratedMap, "FirstName"))
}

// Remove a key-value pair from a map and return the map.
func removeFromMap(userGeneratedMap map[string]string, key string) map[string]string {
	// Remove the key-value pair from the map.
	delete(userGeneratedMap, key)
	// Return the map.
	return userGeneratedMap
}
